plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -177,27 +179,32 @@ fu_plugin_dell_dock_backend_device_added(FuPlugin *plugin, FuDevice *device, GEr
 	if (locker == NULL)
 		return FALSE;
 
+	/* probe extend devices under Usb3.1 Gen 2 Hub */
 	if (fu_device_has_private_flag(FU_DEVICE(hub), FU_DELL_DOCK_HUB_FLAG_HAS_BRIDGE)) {
-		/* only add the device with parent to cache */
-		key = fu_device_get_id(FU_DEVICE(hub));
-		if (fu_plugin_cache_lookup(plugin, key) != NULL) {
-			g_debug("Ignoring already added device %s", key);
-			return TRUE;
-		}
-		/* probe for extended devices */
 		if (!fu_plugin_dell_dock_probe(plugin, FU_DEVICE(hub), error))
 			return FALSE;
-		fu_plugin_cache_add(plugin, key, FU_DEVICE(hub));
 	}
 
-	/* add hub instance id after ec probed */
+	/* process hub devices if ec device is added */
 	devices = fu_plugin_get_devices(plugin);
 	ec_device = fu_plugin_dell_dock_get_ec(devices);
-	if (ec_device != NULL) {
-		guint8 ec_type = fu_dell_dock_get_ec_type(ec_device);
-		fu_dell_dock_hub_add_instance(FU_DEVICE(hub), ec_type);
+	if (ec_device == NULL) {
+		fu_plugin_cache_add(plugin, hub_cache_key, FU_DEVICE(hub));
+		return TRUE;
 	}
+
+	/* determine dock type by ec */
+	ec_type = fu_dell_dock_get_ec_type(ec_device);
+	fu_dell_dock_hub_add_instance(FU_DEVICE(hub), ec_type);
```

Comment:
Should you guard to make sure that you can get a valid base type (`ec_type != 0`)?

Reply:
It should be an integer I think not a pointer